colossalai/shardformer/policies/bert.py

```diff
@@ -152,19 +175,68 @@ class BertModelPolicy(BertPolicy):
 
     def __init__(self) -> None:
         super().__init__()
+        self.layers_per_stage = []
+
+    def get_held_layers(self) -> List[Module]:
+        """Get pipeline layers for current stage."""
+        module = self.model
+        stage_manager = self.pipeline_stage_manager()
+        held_layers = []
+        if self.stage_manager.is_first_stage():
+            held_layers.append(module.embeddings)
+        start_idx, end_idx = self.get_stage_index(self.layers_per_stage, stage_manager.stage)
+        held_layers.extend(module.encoder.layer[start_idx:end_idx])
+        if self.stage_manager.is_last_stage():
+            held_layers.append(module.pooler)
+        return held_layers
+
+    def get_shared_params(self) -> List[Dict[int, Tensor]]:
+        """No shared params in bert model"""
+        return []
+
+    def set_layers_per_stage(self) -> None:
+        """Set layers per stage after set model"""
+        self.layers_per_stage = self.distribute_layers(len(self.model.encoder.layer), self.stage_manager.num_stages)
```

Comment:
Please call this method in `get_held_layers()` directly. An additional attribute is not needed.

Reply:
done